lib/improver/orographic_enhancement.py

```diff
@@ -0,0 +1,583 @@
+                - v dot grad z (wind x topography gradient) >=
+                  self.vgradz_thresh_ms (0.0005 m/s)
+
+            Parameters for calculating upstream contribution:
+                - Maximum range of an upstream cell to contribute to the total
+                  enhancement (self.upstream_range_of_influence_km)
+                - Cloud lifetime (self.cloud_lifetime_s) defines the standard
+                  deviation of the distance weighting function for upstream
+                  enhancement contributions
+                - Scaling factor by which to multiply the weighted sum of
+                  upstream contributions (self.efficiency_factor)
+
+        Create placeholder class members for regridded variable cubes
+        (orography, temperature, humidity, pressure and wind components),
+        saturation vapour pressure, V.gradZ (uplift) array and grid spacing.
+        """
+        self.orog_thresh_m = 20.
+        self.rh_thresh_ratio = 0.8
+        self.vgradz_thresh_ms = 0.0005
+
+        self.upstream_range_of_influence_km = 15.
+        self.cloud_lifetime_s = 102.
+        self.efficiency_factor = 0.23265
+
+        # initialise class members to store regridded variables for
+        # orographic enhancement calculation
+        self.topography = None
+        self.temperature = None
+        self.humidity = None
+        self.pressure = None
+        self.uwind = None
+        self.vwind = None
+
+        # initialise class members for derived variables and metadata
+        self.vgradz = None
+        self.svp = None
+        self.grid_spacing_km = None
+
+    def __repr__(self):
+        """Represent the plugin instance as a string"""
+        return '<OrographicEnhancement()>'
+
+    def _orography_gradients(self):
+        """
+        Calculates the dimensionless gradient of self.topography along both
+        spatial axes, smoothed along the perpendicular axis.  If spatial
+        coordinates are not in the same units as topography height (m),
+        converts coordinate units in place.
+
+        Returns:
+            (tuple): tuple containing:
+                **gradx** (iris.cube.Cube):
+                    2D cube of dimensionless topography gradients in the
+                    positive x direction
+                **grady** (iris.cube.Cube):
+                    2D cube of dimensionless topography gradients in the
+                    positive y direction
+        """
+        self.topography.coord(axis='x').convert_units(self.topography.units)
+        xdim = self.topography.coord_dims(self.topography.coord(axis='x'))[0]
+        self.topography.coord(axis='y').convert_units(self.topography.units)
+        ydim = self.topography.coord_dims(self.topography.coord(axis='y'))[0]
+
+        # smooth topography along perpendicular axis before calculating
+        # gradients in each direction (as done in STEPS)
+        topo_smx = uniform_filter1d(self.topography.data, 3, axis=ydim)
```

Comment:
We're doing all of this on a 1 km grid anyway, which is the resolution of the UKPP topography field.  This was taken straight from how STEPS does this, which is smoothing over one grid cell in each direction.  I can make this a less-magic number if you'd like, but I don't think this will be something we want to configure manually in the short term.

Reply:
Okay, leave it then.